docs/cookbook/create-basic-grid.md

```diff
@@ -0,0 +1,270 @@
+```
+
+### Prepare your data (DataSource)
+`GridFactory::create()` requires some implementation of `DataSourceInterface` as its second argument.
+So we create one, that returns all articles we need.
+
+In our `ArticleGridFactory`, we add a new protected method, that creates and return data source.
+And because we want to get data from the database, we use data source created from Doctrine Query Builder.
+
+```diff
++   /**
++    * @var \Doctrine\ORM\EntityManagerInterface
++    */
++   private $entityManager;
+
+-   public function __construct(GridFactory $gridFactory)
++   public function __construct(GridFactory $gridFactory, EntityManagerInterface $entityManager)
+    {
+        $this->gridFactory = $gridFactory;
++       $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @return \Shopsys\FrameworkBundle\Component\Grid\Grid
+     */
+    public function create(): Grid
+    {
+-       /* @TODO: implement */
++       $grid = $this->gridFactory->create('articleGrid', $this->createAndGetDataSource());
+
++       return $grid;
+    }
+```
+
+A most interesting method is `createAndGetDataSource` that should be in the same `ArticleGridFactory` and will look like this
+```php
+    /**
+     * @return \Shopsys\FrameworkBundle\Component\Grid\DataSourceInterface
+     */
+    protected function createAndGetDataSource(): DataSourceInterface
+    {
+        $queryBuilder = $this->entityManager->createQueryBuilder();
+
+        $queryBuilder->select('a')
+            ->from(Article::class, 'a');
+
+        return new QueryBuilderDataSource($queryBuilder, 'a.id');
+    }
+```
+
+## Present data with Grid
+We prepared our grid, but for now, it is not rendered anywhere and does not contain any columns.
+We change this now.
+
+First, we add columns we want to see into `create` method of `ArticleGridFactory`
+
+```diff
+    public function create(): Grid
+    {
+        $grid = $this->gridFactory->create('articleGrid', $this->createAndGetDataSource());
+
++       $grid->addColumn('id', 'a.id', t('Id'));
++       $grid->addColumn('name', 'a.name', t('Title'));
++       $grid->addColumn('createdAt', 'a.createdAt', t('Created'));
+
+        return $grid;
+    }
+```
+
+*Note: In the example above are column names translated. Do not forget to [dump translations](/docs/introduction/console-commands-for-application-management-phing-targets.md#dump-translations).*
+
+Grid is ready to show all the articles from the database.
+And we just need to render grid itself in a controller.
+For this, we reuse the one created in [Adding a New Administration Page](/docs/cookbook/adding-a-new-administration-page.md) cookbook and add new action method for presenting articles.
+Do not forget to create a twig template for the new action.
+
+We need to inject (pass through constructor) `ArticleGridFactory` created earlier and pass grid view to the template.
+
+```php
+class DashboardController extends AdminBaseController
+{
+    ...
```

Comment:
I would expect here the added dependency on `ArticleGridFactory`, so new property and extended constructor :slightly_smiling_face: